xtask/src/dist.rs

```diff
@@ -212,12 +212,18 @@ fn gather_pgo_profile<'a>(
 }
 
 /// Downloads a crate from GitHub, stores it into `pgo_dir` and returns a path to it.
-fn download_crate_for_training(sh: &Shell, pgo_dir: &Path, url: &str) -> anyhow::Result<PathBuf> {
-    let normalized_path = url.replace("/", "-");
+fn download_crate_for_training(sh: &Shell, pgo_dir: &Path, repo: &str) -> anyhow::Result<PathBuf> {
+    let mut it = repo.splitn(2, '@');
+    let repo = it.next().unwrap();
+    let revision = it.next();
+    let revision =
+        if let Some(revision) = revision { &["--revision", revision] as &[&str] } else { &[] };
```

Comment:
I don't think that `git clone` supports a `--revision` parameter :thinking: I think that `--branch` should work though.

Reply:
Mine does, but you make a good point: https://git-scm.com/docs/git-clone#Documentation/git-clone.txt-code--revisionltrevgtcode. It's a bit better than `--branch` because it also supports commits, not just branches and tags.

I've changed it to `--branch` and added a FIXME.

Reply:
Ah, probably it was added in a later version. I have `2.43.0` and it doesn't support the flag.

Reply:
Yeah, looks like it's from `2.49`, released.. last month :sweat_smile:.